controllers/buildpipeline/buildpipeline_controller.go

```diff
@@ -104,14 +103,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	if err != nil {
 		logger.Error(err, "Failed to get Application from Component",
 			"Component.Name ", component.Name, "Component.Namespace ", component.Namespace)
+		tknErr := tekton.AnnotateBuildPipelineRunWithCreateSnapshotAnnotation(ctx, pipelineRun, r.Client, err)
+		if tknErr != nil {
+			return ctrl.Result{}, tknErr
+		}
+		if errors.IsNotFound(err) {
+			return helpers.HandleLoaderError(logger, err, "application", "component")
```

Comment:
We don't need this if statement here, I noticed that HandleLoaderError is doing it already